packages/core/src/connectors/walletConnect.ts

```diff
@@ -98,9 +110,13 @@ export class WalletConnectConnector extends Connector<
     return chainId
   }
 
-  async getProvider(create?: boolean) {
+  async getProvider(
+    create?: boolean,
+    {
+      chainId = this.options?.chainId || this.chains[0].id,
+    }: { chainId?: number } = {},
+  ) {
```

Comment:
Maybe we should change `getProvider`'s interface from:

```ts
abstract getProvider(create?: boolean): Promise<Provider>
```

To an object:

```ts
abstract getProvider(config: { chainId?: number; create?: boolean } = {}): Promise<Provider>
```

Makes things more flexible in the future if we need to add on more params (maybe for [multi-chain support](https://github.com/tmm/wagmi/discussions/396) eventually).

Reply:
Sounds good!